src/domain/hooks/hooks.repository.ts

```diff
@@ -23,11 +28,19 @@ import { IStakingRepository } from '@/domain/staking/staking.repository.interfac
 import { memoize, MemoizedFunction } from 'lodash';
 
 @Injectable()
-export class HooksRepositoryWithNotifications implements IHooksRepository {
+export class HooksRepositoryWithNotifications
+  implements IHooksRepository, OnModuleInit, OnModuleDestroy
+{
   private static readonly HOOK_TYPE = 'hook';
+  private static readonly UNSUPPORTED_CHAIN_EVENT = 'unsupported_chain_event';
+  public static readonly UNSUPPORTED_EVENTS_LOG_INTERVAL = 60 * 1000; // 1 minute
   private readonly queueName: string;
-  private isSupportedChainMemo: ((chainId: string) => Promise<boolean>) &
+  private readonly isSupportedChainMemo: ((
+    chainId: string,
+  ) => Promise<boolean>) &
     MemoizedFunction;
+  private readonly unsupportedEventsStore: Map<string, number> = new Map();
```

Comment:
As we have two instances and this is stored in memory, won't this mean the log is emitted twice each time?

Reply:
I moved this to Redis in aea854488ca98389b1d3878f84dad1eadcadb337.

Reply:
Thank you for looking into this Aaron! The memory storage was created to alleviate cache usage when receiving a lot of unsupported events in a short period. The log would indeed be emitted N times on each cycle (N being the number of running instances). Still, each instance would print the number of events received by the instance, and monitoring tools can aggregate the numbers if needed. 

Sorry, this approach was highly opinionated and probably should have been discussed beforehand, I'm fine with using Redis for storing a unified counter, but I also think we should use an atomic counter (Redis `INCR`) instead of a plain hash (as implemented on https://github.com/safe-global/safe-client-gateway/commit/aea854488ca98389b1d3878f84dad1eadcadb337) for both consistency (dirty writes) and performance. Wdyt @iamacook @PooyaRaki ?

(Btw, if we think this should be changed, I'd be super happy to tackle it on a separate PR so we can merge this one and unblock the feature 🙂 )